### PR TITLE
Pbl 31966/include user and watch info

### PR DESCRIPTION
**Pull request by keegan-lillo**
This PR exposes a `.meta` property on `ClayConfig` that contains info on the current user and connected watch

### CODE REVIEW

**Comment by keegan-lillo** [comment]
ugh travis is being weird. #17 didn't work :(

**Comment by keegan-lillo** [comment]
Looks like the longer timeout worked. Hopefully it stays that way. I would have liked to have found the actual cause though.

File: test/spec/index.js [hunk 1]
```diff
@@ -56,6 +74,8 @@ describe('Clay', function() {
 
     it('handles the "showConfiguration" event if autoHandleEvents is not false',
     function() {
+      this.timeout(10000); // 10 seconds
```
Comment (chapel) [inline test/spec/index.js]: Have you tried logging/debugging this bit of code in this test to see where it is stopping and where the code paths take you?

Comment (keegan-lillo) [inline test/spec/index.js]: I can't reproduce it locally. Only in Travis. I'll give logging a shot though.

Comment (keegan-lillo) [inline test/spec/index.js]: Adding a log line at every step didn't shed any light. It looks like the test runs through all the instructions but still freaks out. I'm okay with the timeout being the solution.

**Comment by keegan-lillo** [comment]
@chapel Other than that weirdness with the timeout, are you cool with the other changes?

File: index.js [hunk 2]
```diff
@@ -29,11 +29,17 @@ function Clay(config, customFn, options) {
...
+    activeWatchInfo: null,
+    accountToken: '',
+    watchToken: ''
+  };
 
   // Let Clay handle all the magic
   if (options.autoHandleEvents !== false && typeof Pebble !== 'undefined') {
 
     Pebble.addEventListener('showConfiguration', function() {
+      _populateMeta();
```
Comment (chapel) [inline index.js]: Why does this need to be here and not just in `ready`?

File: index.js [hunk 3]
```diff
@@ -49,6 +55,23 @@ function Clay(config, customFn, options) {
         console.log(JSON.stringify(error));
       });
     });
+  } else if (typeof Pebble !== 'undefined') {
```
Comment (chapel) [inline index.js]: What do you do if `Pebble` is `undefined`?

**Comment by chapel** [comment]
Some questions, rest :+1: 

As far as the timeout issue, I don't have a good explanation for it and unless we were to dig more into it, it is hard to debug since it is only on travis. I do think it should be explored but maybe not for this PR.

File: index.js (hunk 2 above)
Comment (keegan-lillo) [inline index.js]: We only use the ready event if the developer is not wanting clay to auto handle the showConfiguration event.

File: index.js (hunk 3 above)
Comment (keegan-lillo) [inline index.js]: We don't do anything. If there is no Pebble global Clay is in a development environment.

**Pull request merged**